cmake/OpenCVFindIPP.cmake

```diff
@@ -34,6 +34,11 @@ unset(IPP_VERSION_MAJOR)
 unset(IPP_VERSION_MINOR)
 unset(IPP_VERSION_BUILD)
 
+if (X86 AND UNIX AND NOT APPLE AND NOT ANDROID AND BUILD_SHARED_LIBS)
+    message(STATUS "On 32-bit Linux IPP can not currently be used with dynamic libs. Set BUILD_SHARED_LIBS=OFF")
```

Comment:
There is a linker issue: [http://code.opencv.org/issues/4070](http://code.opencv.org/issues/4070)
Also some other mentions: 
- https://software.intel.com/en-us/forums/topic/335961
- http://answers.opencv.org/question/37115/opencv-249-make-error/

It can be related with the following note from [IPP 8.1 Release Notes](https://software.intel.com/en-us/articles/intel-ipp-81-library-release-notes):
>Known Intel® IPP 8.1 Issues and Limitations:

>- On Linux* OS, Intel IPP shared libraries for IA-32 contain non-PIC symbols. Each process that uses Intel IPP shared libraries uses a separate copy of the library instead of using one common library.